modules/contact/src/ReferenceResidualProblem.C

```diff
@@ -145,14 +271,30 @@ ReferenceResidualProblem::updateReferenceResidual()
   System & s = nonlinear_sys.system();
   TransientExplicitSystem & as = aux_sys.sys();
 
+  for (unsigned int i = 0; i < _group_resid.size(); ++i)
+  {
+    _group_resid[i] = 0.0;
+    _group_refResid[i] = 0.0;
+  }
+
   for (unsigned int i = 0; i < _solnVars.size(); ++i)
+  {
     _resid[i] = s.calculate_norm(nonlinear_sys.RHS(), _solnVars[i], DISCRETE_L2);
+    _group_resid[_variable_group_num_index[i]] += pow(_resid[i], 2);
```

Comment:
Use something like `libMesh::Utility::pow<2>(_resid[i])` everywhere you are raising things to integer powers.